ingest/mongo_connection.py

```diff
@@ -75,13 +77,25 @@ def wrapper(*args, **kwargs):
                 else:
                     raise e
             except BulkWriteError as bwe:
+
                 if attempt < MAX_ATTEMPTS - 1:
                     dev_logger.warning(
                         "Batch ops error occurred. Reinsert attempt %s.", str(attempt)
                     )
+                    error_docs = bwe.details["writeErrors"]
+                    # Check error code to see if any failures are due to duplicate ids (error code 11000)
+                    if any(doc["code"] == 11000 for doc in error_docs):
+                        args[0] = reassign_id(args[0])
                     retry(attempt)
                 else:
                     dev_logger.debug(str(bwe.details))
                     raise bwe
 
     return wrapper
+
+
+def reassign_id(documents):
```

Comment:
Technically, the error code `11000` is `duplicate key error index`, which means that an index is invalid because we're violating a constraint, usually for uniqueness.  I think all the instances we've run across right now are for the main document ID, but it is possible that there are other unique indexes that are throwing the error.  For instance, there are 3 other indexes for `DataArray` documents, one of which is a unique 10-value compound index that could also throw the same error, but it would not be for the main document ID.

Not a blocking concern for now, but something we should be aware of.